AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginAuthIntegrationTests/AWSDataStoreCategoryPluginAuthIntegrationTests.swift

```diff
@@ -0,0 +1,230 @@
+            try Amplify.add(plugin: AWSCognitoAuthPlugin())
+            let amplifyConfig = try TestConfigHelper.retrieveAmplifyConfiguration(forResource: amplifyConfigurationFile)
+            try Amplify.configure(amplifyConfig)
+
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+        if isSignedIn() {
+            signOut()
+        }
+    }
+
+    override func tearDown() {
+        signOut()
+        Amplify.reset()
+    }
+
+    /// A user can persist data in the local store without signing in. Once the user signs in,
+    /// the sync engine will start and sync the mutations to the cloud. Once the reconcillation is complete, retrieving
+    /// the same data will contain ownerId
+    ///
+    /// - Given: A DataStore plugin configured with SocialNote model containing with auth rules
+    /// - When:
+    ///    - User is not signed in, then user can successfully save a note to local store
+    ///    - User remains signed out, then user can successfully retrieve the saved note, with empty owner field
+    ///    - User signs in, retrieves note, sync engine is started and reconciles local store with the ownerId
+    ///    - The note now it contains the ownerId
+    func testUnauthenticatedSavesToLocalStoreIsReconciledWithCloudStoreAfterAuthentication() throws {
+        let savedLocalNote = saveNote(content: "owner saved note")
+        let queriedNoteOptional = queryNote(byId: savedLocalNote.id)
+        guard let note = queriedNoteOptional else {
+            XCTFail("Faled to query local note")
+            return
+        }
+        XCTAssertNil(note.owner)
+
+        let syncReceivedInvoked = expectation(description: "Received SyncReceived event")
+        var remoteNoteOptional: SocialNote?
+        let syncReceivedListener = Amplify.Hub.listen(to: .dataStore, eventName: syncReceived) { payload in
+            guard let mutationEvent = payload.data as? MutationEvent,
+                let note = try? mutationEvent.decodeModel() as? SocialNote else {
+                    XCTFail("Can't cast payload as mutation event")
+                    return
+            }
+            if note.id == savedLocalNote.id {
+                remoteNoteOptional = note
+                syncReceivedInvoked.fulfill()
+            }
+        }
+        guard try HubListenerTestUtilities.waitForListener(with: syncReceivedListener, timeout: 5.0) else {
+            XCTFail("syncReceivedListener registered for hub")
+            return
+        }
+
+        signIn(username: user1.username, password: user1.password)
+
+        wait(for: [syncReceivedInvoked], timeout: TestCommonConstants.networkTimeout)
+        Amplify.Hub.removeListener(syncReceivedListener)
+        guard let remoteNote = remoteNoteOptional else {
+            XCTFail("Should have received a SyncReceived event with the remote note reconciled to local store")
+            return
+        }
+        XCTAssertNotNil(remoteNote.owner)
```

Comment:
instead of checking for not nil..seems like we should check to make sure the owner is a specific value?

Reply:
adding check against `Auth.getCurrentUser().username`